<?xml version="1.0"?>
<launch>

  <rosparam param="ticks_meter">17</rosparam>
  <node pkg="duda_go" type="motor_run.py" name="run_motor"/>
  
  <node pkg="duda_go" type="pid_velocity.py" name="lpid_velocity">
      <remap from="wheel" to="lwheel"/>
      <remap from="motor_cmd" to="lmotor_cmd"/>
      <remap from="wheel_vtarget" to="lwheel_vtarget"/>
      <remap from="wheel_vel" to="lwheel_vel"/>
      <rosparam param="Kp">15</rosparam>
      <rosparam param="Ki">0</rosparam>
      <rosparam param="Kd">0</rosparam>
      <rosparam param="out_min">0</rosparam>
      <rosparam param="out_max">63</rosparam>
      <rosparam param="rate">30</rosparam>
      <rosparam param="timeout_ticks">10</rosparam>
      <rosparam param="rolling_pts">5</rosparam>
  </node>
  
  <node pkg="duda_go" type="pid_velocity.py" name="rpid_velocity">
      <remap from="wheel" to="rwheel"/>
      <remap from="motor_cmd" to="rmotor_cmd"/>
      <remap from="wheel_vtarget" to="rwheel_vtarget"/>
      <remap from="wheel_vel" to="rwheel_vel"/>
      <rosparam param="Kp">16</rosparam>
      <rosparam param="Ki">0</rosparam>
      <rosparam param="Kd">0</rosparam>
      <rosparam param="out_min">0</rosparam>
      <rosparam param="out_max">63</rosparam>
      <rosparam param="rate">30</rosparam>
      <rosparam param="timeout_ticks">10</rosparam>
      <rosparam param="rolling_pts">5</rosparam>
  </node> 

  <!--node pkg="duda_go" type="virtual_joystick.py" name="virtual_joystick" output="screen"/-->
  <node pkg="duda_go" type="twist_to_motors.py" name="twist_to_motors" output="screen">
    <rosparam param="base_width">0.245</rosparam>
  </node>

  <node pkg="rosserial_python" type="serial_node.py" name="encoder" output="screen" >
    <rosparam param="port">/dev/ttyUSB0</rosparam>
    <rosparam param="baud">2000000</rosparam>
  </node>

  <!-- node pkg="duda_go" type="motor_run.py" name="motor_run" output="screen">
    <rosparam param="port">/dev/ttyUSB0</rosparam>
  </node> -->

</launch>

